File: tests/fixtures/sample.go
// Fixture: Go module for lexical parser tests.
// Includes structs, interfaces, functions, imports, and TODO comments.

package processor

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"os"
)

// TODO: Add connection pooling
// FIXME: Context cancellation not implemented

// DataRecord represents a single record from the data store.
type DataRecord struct {
	ID    string            `json:"id"`
	Value float64           `json:"value"`
	Tags  []string          `json:"tags"`
	Meta  map[string]string `json:"meta"`
}

// Transformer defines the interface for data transformations.
type Transformer interface {
	Transform(record *DataRecord) (*DataRecord, error)
	Validate(record *DataRecord) bool
}

// DataProcessor handles loading and processing of records.
type DataProcessor struct {
	config map[string]string
	logger *os.File
}

// NewDataProcessor creates a new processor with the given config.
func NewDataProcessor(config map[string]string) *DataProcessor {
	return &DataProcessor{config: config}
}

// LoadRecords reads and parses records from a JSON file.
func (p *DataProcessor) LoadRecords(path string) ([]DataRecord, error) {
	// BUG: Large files are read entirely into memory
	data, err := ioutil.ReadFile(path)
	if err != nil {
		return nil, fmt.Errorf("failed to read file: %w", err)
	}
	var records []DataRecord
	if err := json.Unmarshal(data, &records); err != nil {
		return nil, fmt.Errorf("failed to parse JSON: %w", err)
	}
	return records, nil
}

// Process filters and transforms a slice of records.
func (p *DataProcessor) Process(records []DataRecord) []DataRecord {
	// TODO: Apply transformers from config
	var result []DataRecord
	for _, r := range records {
		if r.Value > 0 {
			result = append(result, r)
		}
	}
	return result
}

// RunPipeline is the top-level entry point for the processing pipeline.
func RunPipeline(configPath string) error {
	config := map[string]string{"path": configPath}
	processor := NewDataProcessor(config)
	records, err := processor.LoadRecords(configPath)
	if err != nil {
		return err
	}
	processor.Process(records)
	return nil
}

File: tests/fixtures/sample.ts
// Fixture: TypeScript module for lexical parser tests.
// Includes interfaces, classes, functions, imports, and TODO comments.

import { readFileSync } from "fs";
import axios from "axios";
import type { Config } from "./types";

// TODO: Add request retry logic
// FIXME: Error handling is incomplete

export interface DataPayload {
  id: string;
  records: Record<string, unknown>[];
}

export class ApiClient {
  private baseUrl: string;

  constructor(config: Config) {
    this.baseUrl = config.apiUrl;
  }

  async fetchData(endpoint: string): Promise<DataPayload> {
    // BUG: timeout is not configurable
    const response = await axios.get(`${this.baseUrl}/${endpoint}`);
    return response.data as DataPayload;
  }

  async postData(endpoint: string, payload: unknown): Promise<void> {
    await axios.post(`${this.baseUrl}/${endpoint}`, payload);
  }
}

export class DataTransformer {
  transform(payload: DataPayload): DataPayload {
    return {
      ...payload,
      records: payload.records.filter((r) => Object.keys(r).length > 0),
    };
  }
}

export function loadConfig(filePath: string): Config {
  const raw = readFileSync(filePath, "utf-8");
  return JSON.parse(raw) as Config;
}

export async function runPipeline(configPath: string): Promise<void> {
  const config = loadConfig(configPath);
  const client = new ApiClient(config);
  const data = await client.fetchData("records");
  const transformer = new DataTransformer();
  transformer.transform(data);
}
